File: output/GPT4o-Mini/xCodeEval_remove_4000/C++/Go/0b1a54afd90ac7191f4cccd573d4648c.go
package main

import (
	"fmt"
	"math/rand"
	"time"
)

type lli int64

func ctoi(c rune) int {
	if c >= '0' && c <= '9' {
		return int(c - '0')
	}
	return -1
}

func alphabetPos(c rune) int {
	if c >= 'a' && c <= 'z' {
		return int(c - 'a')
	}
	return -1
}

func id6(c rune) int {
	if c >= 'A' && c <= 'Z' {
		return int(c - 'A')
	}
	return -1
}

func split(str string, ch rune) []string {
	var result []string
	first := 0
	last := 0

	for {
		last = first
		for last < len(str) && rune(str[last]) != ch {
			last++
		}
		result = append(result, str[first:last])
		if last >= len(str) {
			break
		}
		first = last + 1
	}
	return result
}

func gcd(a, b int) int {
	if a < b {
		a, b = b, a
	}
	if b == 0 {
		return a
	}
	if a%b == 0 {
		return b
	}
	return gcd(b, a%b)
}

func gcdLong(a, b lli) lli {
	if a < b {
		a, b = b, a
	}
	if b == 0 {
		return a
	}
	if a%b == 0 {
		return b
	}
	return gcdLong(b, a%b)
}

func lcm(a, b int) int {
	return a * b / gcd(a, b)
}

func lcmLong(a, b lli) lli {
	return a * b / gcdLong(a, b)
}

func powFast(x, nPower, modulus lli) lli {
	if nPower == 0 {
		return 1
	}
	if nPower%2 == 0 {
		return powFast((x*x)%modulus, nPower/2, modulus)
	}
	return (x * powFast(x, nPower-1, modulus)) % modulus
}

type CombinationTable struct {
	val [][]lli
}

func NewCombinationTable(size int) *CombinationTable {
	ct := &CombinationTable{
		val: make([][]lli, size+1),
	}
	for i := range ct.val {
		ct.val[i] = make([]lli, size+1)
	}
	for i := 0; i <= size; i++ {
		for j := 0; j <= i; j++ {
			if j == 0 || j == i {
				ct.val[i][j] = 1
			} else {
				ct.val[i][j] = ct.val[i-1][j-1] + ct.val[i-1][j]
			}
		}
	}
	return ct
}

func printVector(h []int, verbose bool) {
	L := len(h)
	for i := 0; i < L; i++ {
		if verbose {
			fmt.Printf("%d", h[i])
		} else {
			fmt.Print(h[i])
		}
		if i != L-1 {
			if verbose {
				fmt.Print(" ")
			} else {
				fmt.Print(" ")
			}
		} else {
			if verbose {
				fmt.Println()
			} else {
				fmt.Println()
			}
		}
	}
}

func printVectorLong(h []lli, verbose bool) {
	L := len(h)
	for i := 0; i < L; i++ {
		if verbose {
			fmt.Printf("%d", h[i])
		} else {
			fmt.Print(h[i])
		}
		if i != L-1 {
			if verbose {
				fmt.Print(" ")
			} else {
				fmt.Print(" ")
			}
		} else {
			if verbose {
				fmt.Println()
			} else {
				fmt.Println()
			}
		}
	}
}

func id7(h [][]int, verbose bool) {
	Ly := len(h)
	for i := 0; i < Ly; i++ {
		Lx := len(h[i])
		for j := 0; j < Lx; j++ {
			if verbose {
				fmt.Printf("%d", h[i][j])
			} else {
				fmt.Print(h[i][j])
			}
			if j != Lx-1 {
				if verbose {
					fmt.Print(" ")
				} else {
					fmt.Print(" ")
				}
			} else {
				if verbose {
					fmt.Println()
				} else {
					fmt.Println()
				}
			}
		}
	}
}

func id7Long(h [][]lli, verbose bool) {
	Ly := len(h)
	for i := 0; i < Ly; i++ {
		Lx := len(h[i])
		for j := 0; j < Lx; j++ {
			if verbose {
				fmt.Printf("%d", h[i][j])
			} else {
				fmt.Print(h[i][j])
			}
			if j != Lx-1 {
				if verbose {
					fmt.Print(" ")
				} else {
					fmt.Print(" ")
				}
			} else {
				if verbose {
					fmt.Println()
				} else {
					fmt.Println()
				}
			}
		}
	}
}

func id7String(h []string, verbose bool) {
	Ly := len(h)
	for i := 0; i < Ly; i++ {
		Lx := len(h[i])
		for j := 0; j < Lx; j++ {
			if verbose {
				fmt.Printf("%c", h[i][j])
			} else {
				fmt.Print(h[i][j])
			}
		}
		if verbose {
			fmt.Println()
		} else {
			fmt.Println()
		}
	}
}

func printBinary(val int, numDigit int, verbose bool) {
	for k := numDigit - 1; k >= 0; k-- {
		if verbose {
			fmt.Printf("%d", (val>>k)&1)
		} else {
			fmt.Print((val >> k) & 1)
		}
	}
	if verbose {
		fmt.Println()
	} else {
		fmt.Println()
	}
}

func printBinaryLong(val lli, numDigit int, verbose bool) {
	for k := numDigit - 1; k >= 0; k-- {
		if verbose {
			fmt.Printf("%d", (val>>k)&1)
		} else {
			fmt.Print((val >> k) & 1)
		}
	}
	if verbose {
		fmt.Println()
	} else {
		fmt.Println()
	}
}

type SegmentTree struct {
	n    int
	node []lli
	id0  lli
}

func NewSegmentTree(v []lli, id3 lli) *SegmentTree {
	sz := len(v)
	n := 1
	for n < sz {
		n <<= 1
	}
	node := make([]lli, 2*n-1)
	for i := 0; i < sz; i++ {
		node[i+n-1] = v[i]
	}
	for i := n - 2; i >= 0; i-- {
		node[i] = min(node[2*i+1], node[2*i+2])
	}
	return &SegmentTree{n: n, node: node, id0: id3}
}

func (st *SegmentTree) update(x int, val lli) {
	x += (st.n - 1)
	st.node[x] = val
	for x > 0 {
		x = (x - 1) / 2
		st.node[x] = min(st.node[2*x+1], st.node[2*x+2])
	}
}

func (st *SegmentTree) getMin(a, b int, k, l, r int) lli {
	if r < 0 {
		r = st.n
	}
	if r <= a || b <= l {
		return st.id0
	}
	if a <= l && r <= b {
		return st.node[k]
	}
	vl := st.getMin(a, b, 2*k+1, l, (l+r)/2)
	vr := st.getMin(a, b, 2*k+2, (l+r)/2, r)
	return min(vl, vr)
}

type id1 struct {
	size_ int
	data  []lli
}

func NewId1(sz int, id2 lli) *id1 {
	data := make([]lli, sz+1)
	for i := range data {
		data[i] = id2
	}
	return &id1{size_: sz, data: data}
}

func (i *id1) sum(idx int) lli {
	if idx <= 0 {
		return 0
	}
	if idx > i.size_ {
		idx = i.size_
	}
	sm := lli(0)
	for idx > 0 {
		sm += i.data[idx]
		idx -= idx & -idx
	}
	return sm
}

func (i *id1) add(idx int, x lli) {
	if idx < 0 || idx >= i.size_ {
		return
	}
	idx++
	for idx <= i.size_ {
		i.data[idx] += x
		idx += idx & -idx
	}
}

type RollingHash struct {
	hash     []lli
	basePow  []lli
	base     lli
	modulus  lli
	id5     lli
	MASK30   lli
	MASK31   lli
	BASE_MIN lli
}

func NewRollingHash(S string, base_ lli, id4 lli) *RollingHash {
	id5 := (1 << 61) - 1
	MASK30 := (1 << 30) - 1
	MASK31 := (1 << 31) - 1
	BASE_MIN := int64(1e7)

	rh := &RollingHash{
		id5:     id5,
		MASK30:  MASK30,
		MASK31:  MASK31,
		BASE_MIN: BASE_MIN,
	}
	if id4 > 0 {
		rh.modulus = id4
	} else {
		rh.modulus = id5
	}
	if base_ > 0 {
		rh.base = base_
	} else {
		rand.Seed(time.Now().UnixNano())
		rh.base = rand.Int63n(id5-BASE_MIN) + BASE_MIN
	}
	if len(S) > 0 {
		rh.initialize(S)
	}
	return rh
}

func (rh *RollingHash) initialize(S string) {
	N := len(S)
	s := make([]int, N)
	for i := 0; i < N; i++ {
		s[i] = int(S[i])
	}
	rh.initializeInt(s)
}

func (rh *RollingHash) initializeInt(S []int) {
	hash := make([]lli, len(S)+1)
	basePow := make([]lli, len(S)+1)
	hash[0] = 0
	basePow[0] = 1

	if rh.modulus == rh.id5 {
		for i := 1; i <= len(S); i++ {
			hash[i] = rh.Modulus2Pow61m1(rh.Multiple2Pow61m1(hash[i-1], rh.base) + lli(S[i-1]))
			basePow[i] = rh.Multiple2Pow61m1(basePow[i-1], rh.base)
		}
	} else {
		for i := 1; i <= len(S); i++ {
			hash[i] = (hash[i-1]*rh.base + lli(S[i-1])) % rh.modulus
			basePow[i] = (basePow[i-1] * rh.base) % rh.modulus
		}
	}
	rh.hash = hash
	rh.basePow = basePow
}

func (rh *RollingHash) Modulus2Pow61m1(val lli) lli {
	val = (val & rh.id5) + (val >> 61)
	if rh.id5 < val {
		val -= rh.id5
	}
	return val
}

func (rh *RollingHash) Multiple2Pow61m1(a, b lli) lli {
	au := a >> 31
	ad := a & rh.MASK31
	bu := b >> 31
	bd := b & rh.MASK31

	mid := ad * bu + au * bd
	midu := mid >> 30
	midd := mid & rh.MASK30

	return rh.Modulus2Pow61m1(((au * bu) << 1) + midu + (midd << 31) + ad*bd)
}

func (rh *RollingHash) between(a, b int) lli {
	if rh.modulus == rh.id5 {
		return rh.Modulus2Pow61m1(rh.modulus + rh.hash[b] - rh.Multiple2Pow61m1(rh.hash[a], rh.basePow[b-a]))
	}
	return (rh.modulus + rh.hash[b] - (rh.hash[a] * rh.basePow[b-a]) % rh.modulus) % rh.modulus
}

func main() {
	var N int
	fmt.Scan(&N)
	a := make([]lli, N)
	mp := make(map[lli]int)
	for i := 0; i < N; i++ {
		fmt.Scan(&a[i])
		mp[a[i]]++
	}
	b := make([]lli, N)
	for i := 0; i < N; i++ {
		fmt.Scan(&b[i])
	}

	var vs []lli
	for elem, count := range mp {
		if count > 1 {
			vs = append(vs, elem)
		}
	}

	ans := lli(0)
	for i := 0; i < N; i++ {
		isok := false
		for _, elem := range vs {
			if (elem & a[i]) == a[i] {
				isok = true
				break
			}
		}
		if isok {
			ans += b[i]
		}
	}

	fmt.Println(ans)
}

File: output/GPT3.5-Turbo/xCodeEval_remove_4000/C++/C/0b1a54afd90ac7191f4cccd573d4648c.c
#include <stdio.h>
#include <stdlib.h>
#include <stdbool.h>
#include <string.h>

typedef long long lli;

int ctoi(char c)
{
  if( c >= '0' && c <= '9' )
  {
    return (int)(c-'0');
  }

  return -1;
}

int alphabet_pos(char c) 
{
  if( c >= 'a' && c <= 'z' )
  {
    return (int)(c-'a');
  }

  return -1;
}

int id6(char c)
{
  if( c >= 'A' && c <= 'Z' )
  {
    return (int)(c-'A');
  }

  return -1;
}

int gcd( int a , int b ) 
{
  if( a < b )
  {
    int temp = a;
    a = b;
    b = temp;
  }

  if( b == 0 )
  {
    return a;
  }
  
  if( a % b == 0 )
  {
    return b;    
  }

  return gcd( b , a % b );
}

long long gcd( long long a , long long b ) 
{
  if( a < b )
  {
    long long temp = a;
    a = b;
    b = temp;
  }

  if( b == 0LL )
  {
    return a;
  }
  
  if( a % b == 0 )
  {
    return b;
  }

  return gcd( b , a % b );
}

int lcm( int a , int b ) 
{
  return  a * b / gcd( a , b );
}

long long lcm( long long a , long long b ) 
{
  return  a * b / gcd( a , b );
}

long long pow_fast( long long x, long long n_power , long long modulus )
{
  if( n_power == 0 )
  {
    return 1;
  }
  
  if( n_power % 2 == 0)
  {
    return pow_fast( x * x % modulus , n_power / 2 , modulus );  
  }

  return x * pow_fast( x , n_power - 1 , modulus ) % modulus;
}

struct CombinationTable
{
  long long** val;
  int size;

  CombinationTable( int size ) 
  {
    this->size = size;
    val = (long long**)malloc((size+1) * sizeof(long long*));
    for( int i = 0 ; i <= size ; ++ i ) 
    {
      val[i] = (long long*)malloc((size+1) * sizeof(long long));
    }

    for( int i = 0 ; i <= size ; ++ i ) 
    {
      for( int j = 0 ; j <= i ; ++ j )
      {
        if( j == 0 || j == i )
        {
          val[i][j] = 1LL;
        }
        else
        {
          val[i][j] = val[i-1][j-1] + val[i-1][j];
        }
      }
    }
  }
};

void print_vector(int* h, int L, bool verbose)
{
  for(int i = 0; i < L; ++ i)
  {
    if( verbose )
    {
      printf("%d",h[i]);
    }
    else
    {
      fprintf(stderr, "%d", h[i]);
    }

    if( i != L-1 )
    {
      if( verbose )
      {
        printf(" ");
      }
      else
      {
        fprintf(stderr, " ");
      }
    }
    else
    {
      if( verbose )
      {
        printf("\n");
      }
      else
      {
        fprintf(stderr, "\n");
      }
    }
  }    
}

void print_vector(lli* h, int L, bool verbose)
{
  for(int i = 0; i < L; ++ i)
  {
    if( verbose )
    {
      printf("%lld",h[i]);
    }
    else
    {
      fprintf(stderr, "%lld", h[i]);
    }

    if( i != L-1 )
    {
      if( verbose )
      {
        printf(" ");
      }
      else
      {
        fprintf(stderr, " ");
      }
    }
    else
    {
      if( verbose )
      {
        printf("\n");
      }
      else
      {
        fprintf(stderr, "\n");
      }
    }
  }    
}

void id7(int** h, int Ly, int Lx, bool verbose)
{
  for(int i = 0; i < Ly; ++ i)
  {
    for(int j = 0; j < Lx; ++ j)
    {	  
      if( verbose )
      {
        printf("%d",h[i][j]);
      }
      else
      {
        fprintf(stderr, "%d", h[i][j]);
      }

      if( j != Lx-1 )
      {
        if( verbose )
        {
          printf(" ");
        }
        else
        {
          fprintf(stderr, " ");
        }
      }
      else
      {
        if( verbose )
        {
          printf("\n");
        }
        else
        {
          fprintf(stderr, "\n");
        }
      }	  
    }
  }    
}

void id7(lli** h, int Ly, int Lx, bool verbose)
{
  for(int i = 0; i < Ly; ++ i)
  {
    for(int j = 0; j < Lx; ++ j)
    {
      if( verbose )
      {
        printf("%lld",h[i][j]);
      }
      else
      {
        fprintf(stderr, "%lld", h[i][j]);
      }

      if( j != Lx-1 )
      {
        if( verbose )
        {
          printf(" ");
        }
        else
        {
          fprintf(stderr, " ");
        }
      }
      else
      {
        if( verbose )
        {
          printf("\n");
        }
        else
        {
          fprintf(stderr, "\n");
        }
      }
    }
  }
}

void id7(char** h, int Ly, bool verbose)
{
  for(int i = 0; i < Ly; ++ i)
  {
    int Lx = strlen(h[i]);
    for(int j = 0; j < Lx; ++ j)
    {
      if( verbose )
      {
        printf("%c",h[i][j]);
      }
      else
      {
        fprintf(stderr, "%c", h[i][j]);
      }
    }
  
    if( verbose )
    {
      printf("\n");
    }
    else
    {
      fprintf(stderr, "\n");
    }
  }    
}

void print_binary(int val, int num_digit, bool verbose)
{
  for(int k = num_digit-1; k >= 0; -- k)
  {
    if( verbose )
    {
      printf("%d",(val>>k)&1);
    }
    else
    {
      fprintf(stderr, "%d", ((val>>k)&1));
    }    
  }
  
  if( verbose )
  {
    printf("\n");
  }
  else
  {
    fprintf(stderr, "\n");
  }
}

void print_binary(lli val, int num_digit, bool verbose)
{
  for(int k = num_digit-1; k >= 0; -- k)
  {
    if( verbose )
    {
      printf("%lld",((val>>k)&1));
    }
    else
    {
      fprintf(stderr, "%lld", ((val>>k)&1));
    }    
  }
  
  if( verbose )
  {
    printf("\n");
  }
  else
  {
    fprintf(stderr, "\n");
  }
}

typedef struct SegmentTree 
{
  int n;
  lli* node;
  lli id0;
} SegmentTree;

SegmentTree* createSegmentTree(int* v, int size, lli id3)
{
  SegmentTree* st = (SegmentTree*)malloc(sizeof(SegmentTree));
  st->n = 1;
  while(st->n < size)
  {
    st->n <<= 1;
  }
  
  st->node = (lli*)malloc((2*st->n-1) * sizeof(lli));
  st->id0 = id3;

  for( int i = 0 ; i < size ; ++ i )
  {
    st->node[i+st->n-1] = v[i];
  }
    
  for( int i = st->n-2 ; i >= 0 ; -- i )
  {
    st->node[i] = (st->node[2*i+1] < st->node[2*i+2]) ? st->node[2*i+1] : st->node[2*i+2];
  } 

  return st;
}

void update(SegmentTree* st, int x, lli val)
{
  x += (st->n - 1);
  st->node[x] = val;
    
  while( x > 0 )
  {
    x = (x - 1) / 2;
    st->node[x] = (st->node[2*x+1] < st->node[2*x+2]) ? st->node[2*x+1] : st->node[2*x+2];
  }
}

lli getmin(SegmentTree* st, int a, int b, int k, int l, int r)
{
  if( r < 0 )
  {
    r = st->n;
  } 
  if( r <= a || b <= l )
  {
    return st->id0;
  } 
  if( a <= l && r <= b )
  {
    return st->node[k];
  }
    
  lli vl = getmin(st, a, b, 2*k+1, l, (l+r)/2);
  lli vr = getmin(st, a, b, 2*k+2, (l+r)/2, r);
  return (vl < vr) ? vl : vr;
}

typedef struct id1 
{
  int size_;
  lli* data;
} id1;

id1* create_id1(int sz, lli id2)
{
  id1* id = (id1*)malloc(sizeof(id1));
  id->size_ = sz;
  id->data = (lli*)malloc((sz+1) * sizeof(lli));
  for(int i = 0; i <= sz; ++i)
  {
    id->data[i] = id2;
  }

  return id;
}

lli sum(id1* id, int i)
{
  if( i <= 0 )
  {
    return (lli) 0.0;
  }

  if( i > id->size_ )
  {
    i = id->size_;
  }
    
  lli sm = 0.0;
  while( i > 0 )
  {
    sm += id->data[i];
    i -= i & -i;
  }
    
  return sm;
}

void add(id1* id, int i, lli x)
{
  if( i < 0 || i >= id->size_ )
  {
    return;
  }
  ++ i;
  while( i <= id->size_ )
  {
    id->data[i] += x;
    i += i & -i;
  }
}

typedef struct RollingHash
{
  unsigned long long* hash;
  unsigned long long* base_pow;
  unsigned long long base, modulus;
  const unsigned long long id5;
  const unsigned long long MASK30;
  const unsigned long long MASK31;
  const unsigned long long BASE_MIN;
} RollingHash;

RollingHash* createRollingHash(char* S, unsigned long long base_, unsigned long long id4)
{
  RollingHash* rh = (RollingHash*)malloc(sizeof(RollingHash));
  rh->id5 = (1ULL << 61) - 1;
  rh->MASK30 = (1ULL << 30) - 1;
  rh->MASK31 = (1ULL << 31) - 1;
  rh->BASE_MIN = 1e7;

  if( 0 < id4 )
  {
    rh->modulus = id4;
  }
  else
  {
    rh->modulus = rh->id5;
  }

  if( 0 < base_ )
  {
    rh->base = base_;
  }
  else
  {
    rh->base = rh->BASE_MIN;
  }

  if( strlen(S) > 0 )
  {
    initialize(S);
  }

  return rh;
}

void initialize(RollingHash* rh, char* S)
{
  int N = strlen(S);
  int* s = (int*)malloc(N * sizeof(int));
  for(int i = 0; i < N; ++i)
  {
    s[i] = S[i];
  }
    
  initialize(rh, s);
}

void initialize(RollingHash* rh, int* S)
{
  int N = strlen(S);
  rh->hash = (unsigned long long*)malloc((N+1) * sizeof(unsigned long long));
  rh->base_pow = (unsigned long long*)malloc((N+1) * sizeof(unsigned long long));
  rh->hash[0] = 0;
  rh->base_pow[0] = 1;

  if( rh->modulus == rh->id5 )
  {	
    for(int i = 1; i <= N; ++i)
    {
      rh->hash[i] = Modulus_2pow61m1( Multiple_2pow61m1(rh->hash[i-1],rh->base) + S[i-1] );
      rh->base_pow[i] = Multiple_2pow61m1(rh->base_pow[i-1],rh->base);
    }
  }
  else
  {
    for(int i = 1; i <= N; ++i)
    {
      rh->hash[i] = ( rh->hash[i-1] * rh->base + S[i-1] ) % rh->modulus;
      rh->base_pow[i] = ( rh->base_pow[i-1] * rh->base ) % rh->modulus;
    }
  }
}

unsigned long long between(RollingHash* rh, int a, int b)
{
  if( rh->modulus == rh->id5 )
  {
    return Modulus_2pow61m1( rh->modulus + rh->hash[b] - Multiple_2pow61m1(rh->hash[a],rh->base_pow[b-a]) );
  }
    
  return ( rh->modulus + rh->hash[b] - ( rh->hash[a] * rh->base_pow[b-a] ) % rh->modulus ) % rh->modulus;
}

int N;
lli* a;
lli* b;

int main()
{
  scanf("%d", &N);
  a = (lli*)malloc(N * sizeof(lli));
  int* mp = (int*)calloc(N, sizeof(int));
  for(int i = 0; i < N; ++i)
  {
    scanf("%lld", &a[i]);
    ++ mp[a[i]];
  }
  b = (lli*)malloc(N * sizeof(lli));
  for(int i = 0; i < N; ++i)
  {
    scanf("%lld", &b[i]);
  }

  lli* vs = (lli*)malloc(N * sizeof(lli));
  int vs_size = 0;
  for( int i = 0 ; i < N ; ++ i )
  {
    if( mp[a[i]] > 1 )
    {
      vs[vs_size++] = a[i];
    }
  }

  lli ans = 0;
  for(int i = 0; i < N; ++i)
  {
    bool isok = false;
    for( int j = 0 ; j < vs_size ; ++ j )
    {
      if( ( vs[j] & a[i] ) == a[i] )
      {
        isok = true;
        break;
      }
    }

    if( isok )
    {
      ans += b[i];
    }
  }

  printf("%lld\n", ans);

  free(a);
  free(b);
  free(mp);
  free(vs);

  return 0;
}
